File: client/src/app/_models/admin/visaHeader.ts
import { IVisaItem, VisaItem } from "./visaItem";

export interface IVisaHeader
{
    id: number;
    visaNo: string;
    customerId: number;
    customerName: string;
    visaDateG: Date;
    visaDateH: string;
    visaExpiryH: string;
    visaExpiryG: Date;
    visaSponsorName: string;
    visaItems: IVisaItem[];
}

export class VisaHeader implements IVisaHeader
{
    id = 0;
    visaNo = '';
    customerId = 0;
    customerName = '';
    visaDateG =  new Date();
    visaDateH = '';
    visaExpiryH = '';
    visaExpiryG = new Date();
    visaSponsorName = '';
    visaItems:IVisaItem[]= [];
}

File: api/Entities/Admin/VisaHeader.cs
using System.ComponentModel.DataAnnotations;
using DocumentFormat.OpenXml.Wordprocessing;

namespace api.Entities.Admin
{
    public class VisaHeader:BaseEntity
    {
        [MaxLength(15), Required]
        public string VisaNo { get; set; }
        public string VisaDateH  { get; set; }   //**TODO** USE hiJRIDATE types
        public DateTime VisaDateG { get; set; }
        [Required]
        public int CustomerId   { get; set; }
        [MaxLength(50)]
        public string CustomerName { get; set; }
        public string VisaExpiryH { get; set; }
        public DateTime VisaExpiryG { get; set; }
        [MaxLength(40), Required]
        public string VisaSponsorName { get; set; }
        public ICollection<VisaItem> VisaItems { get; set; }
    }   

}